tests/integration/targets/test_proxysql_replication_hostgroups/tasks/main.yml

```
@@ -82,6 +82,120 @@
     test_proxysql_replication_hostgroups_with_delayed_persist: true
     test_proxysql_replication_hostgroups_check_idempotence: true
 
+- name: "gather proxysql informations"
+  community.proxysql.proxysql_info:
+    login_user: admin
+    login_password: admin
+  register: proxysql_information
+
+- name: create replication hostgroup
+  community.proxysql.proxysql_replication_hostgroups:
+    login_user: admin
+    login_password: admin
+    writer_hostgroup: 100
+    reader_hostgroup: 200
+  register: special_condition
+
+- name: verify change
+  assert:
+    that:
+      - special_condition is changed
+
+- name: >
+    test conditions that are failing in the past
+  block:
+    - name: change reader hostgroup
+      community.proxysql.proxysql_replication_hostgroups:
+        login_user: admin
+        login_password: admin
+        writer_hostgroup: 100
+        reader_hostgroup: 300
+      register: special_condition
+
+    - name: verify change
+      assert:
+        that: 
+          - special_condition is changed
+
+    - name: update check_type (check_mode)
+      community.proxysql.proxysql_replication_hostgroups:
+        login_user: admin
+        login_password: admin
+        writer_hostgroup: 100
+        reader_hostgroup: 300
+        check_type: innodb_read_only
+      register: special_condition
+      check_mode: yes
+
+    - name: verify check_mode change for proxysql > 2
+      when: proxysql_information.version.major == 2
+      assert:
+        that: 
+          - special_condition is changed
+
+    - name: update check_type
+      community.proxysql.proxysql_replication_hostgroups:
+        login_user: admin
+        login_password: admin
+        writer_hostgroup: 100
+        reader_hostgroup: 300
+        check_type: innodb_read_only
+      register: special_condition
+
+    - name: verify change for proxysql > 2
+      when: proxysql_information.version.major == 2
```

```suggestion
      when: proxysql_information.version.major >= 2
```

how about adding a check `when: proxysql_information.version.major < 2` ? Can we check, e.g., that `- special_condition is not changed`?

---

Yes. I will add

---

how about changing the line as suggested same as you did a bit above?:)